### PR TITLE
Feature/configure email notifications

### PR DESCRIPTION
**Pull request by jblance**
My initial attempt to implement email notification configuring (for issue #8 )

Note: authenticated SMTP is not implemented
Note2: no spec tests as I have figured them out yet....

### CODE REVIEW

**Comment by TraGicCode** [comment]
Hey there,

Great to see you opening issues and pull requests.  Hopefully we can knock some of these out.

I will take a look at this in the morning and provide any feedback that I have.

File: manifests/init.pp [hunk 1]
```diff
@@ -19,6 +19,20 @@
...
+  Enum['Weekly', 'Daily'] $notification_frequency    = $wsusserver::params::notification_frequency,
+  String $notification_time_of_day                   = $wsusserver::params::notification_time_of_day,
+  String $smtp_hostname                              = $wsusserver::params::smtp_hostname,
+  Integer $smtp_port                                 = $wsusserver::params::smtp_port,
+  Boolean $smtp_requires_authentication              = $wsusserver::params::smtp_requires_authentication,
+  String $smtp_username                              = $wsusserver::params::smtp_username,
+  String $smtp_sender_displayname                    = $wsusserver::params::smtp_sender_displayname,
+  String $smtp_sender_emailaddress                   = $wsusserver::params::smtp_sender_emailaddress,
+  String $email_language                             = $wsusserver::params::email_language,
+
```
Comment (TraGicCode) [inline manifests/init.pp]: Remove this extra newline.

File: manifests/init.pp [hunk 2]
```diff
@@ -19,6 +19,20 @@
   Boolean $synchronize_automatically                 = $wsusserver::params::synchronize_automatically,
   String $synchronize_time_of_day                    = $wsusserver::params::synchronize_time_of_day,
   Integer $number_of_synchronizations_per_day        = $wsusserver::params::number_of_synchronizations_per_day,
+  Boolean $send_sync_notification                    = $wsusserver::params::send_sync_notification,
```
Comment (TraGicCode) [inline manifests/init.pp:22]: Can you document these parameters in the README.md ( https://github.com/TraGicCode/tragiccode-wsusserver/blob/master/README.md#wsusserver-1 ) like the other parameters that make up the public interface to this class?

**Review by TraGicCode — CHANGES_REQUESTED** [review]
Looks good so far.  I've requested some minor changes.  Sorry for taking to long to review this.  I have been really busy on my end so i've had to push this down on my list of things to do.

**Comment by TraGicCode** [comment]
Also if you can please squash you commits to clean up some of the noise with the WIP and oops commits.

**Comment by jblance** [comment]
Hi

I made review changes
I've tried to squash the commits, I cant seem to do it except to a new branch? Do you know how, or should I just PR the new branch instead?

Regards

**Comment by TraGicCode** [comment]
This should handle the squash for you.

```powershell
git checkout feature/configure-emailnotifications
git reset $(git merge-base master feature/configure-emailnotifications)
git add -A
git commit -m "Added email feature or something like this"
git push <your_remote_name> HEAD --force
```

**Comment by jblance** [comment]
Thanks for the squash help
Done now - hopefully nothing got undone with my messing around..

File: manifests/init.pp (hunk 1 above)
Comment (TraGicCode) [reply manifests/init.pp]: Still waiting on you to do this.

File: manifests/init.pp (hunk 2 above)
Comment (TraGicCode) [reply manifests/init.pp:22]: Still waiting on you to do this.

**Comment by TraGicCode** [comment]
@jblance The squash looks good to me.  Can you fix the above changes i requested?  Once done simply  type "Done." in each comment.

File: manifests/init.pp (hunk 2 above)
Comment (jblance) [reply manifests/init.pp]: Done

File: manifests/init.pp (hunk 1 above)
Comment (jblance) [reply manifests/init.pp]: Done

File: manifests/init.pp (hunk 2 above)
Comment (jblance) [reply manifests/init.pp]: Oh, sorry now i see where you mean

Comment (jblance) [reply manifests/init.pp:22]: Done

File: manifests/config.pp [hunk 3]
```diff
@@ -36,6 +50,205 @@
...
     }
 
+    # Sync Notification Settings
+    exec { 'wsus-config-sync-notification-settings':
+      #Set sync to true or false
+      command   => "\$ErrorActionPreference = \"Stop\"
+                    \$wsusEmailNotificationConfiguration = (Get-WsusServer).GetEmailNotificationConfiguration()
+                    \$wsusEmailNotificationConfiguration.SendSyncNotification = \$${send_sync_notification}
+                    \$wsusEmailNotificationConfiguration.Save()
+                    return 'SendSyncNotification setting updated'
```
Comment (TraGicCode) [inline manifests/config.pp]: Can we remove this?  Doubt it's needed.

File: manifests/config.pp [hunk 4]
```diff
@@ -36,6 +50,205 @@
...
+
+
+    # Status Report Notification Settings
+    exec { 'wsus-config-status-report-notification-settings':
+      #Set sync to true or false
+      command   => "\$ErrorActionPreference = \"Stop\"
+                    \$wsusEmailNotificationConfiguration = (Get-WsusServer).GetEmailNotificationConfiguration()
+                    \$wsusEmailNotificationConfiguration.SendStatusNotification = \$${send_status_notification}
+                    \$wsusEmailNotificationConfiguration.Save()
+                    return 'SendStatusNotification setting updated'
```
Comment (TraGicCode) [inline manifests/config.pp]: Same here.  See above.

**Review by TraGicCode — CHANGES_REQUESTED** [review]

**Comment by TraGicCode** [comment]
@jblance Here are some more minor changes.  I will run setup some beaker tests and perform some acceptance testing and let you know of any issues i run into.

File: manifests/config.pp (hunk 3 above)
Comment (jblance) [reply manifests/config.pp]: Remove which bit?

This sets the "Send email on sync update", i.e turning the feature on or off
I think this is needed, unless I'm missing something

Comment (TraGicCode) [reply manifests/config.pp]: I'm referring to the `return 'SendSyncNotification setting updated'` piece since it is not necessary

Comment (jblance) [reply manifests/config.pp]: Oh OK - I've removed them 

I did quite like that it output that info when running `puppet agent -tv` manually

File: manifests/config.pp (hunk 4 above)
Comment (jblance) [reply manifests/config.pp]: Done

**Review by TraGicCode — APPROVED** [review]

**Comment by TraGicCode** [comment]
@jblance Thank you for your contribution.  Hope to see more again soon!

**Pull request merged**